### DISCUSSION TITLE
Custom Image node with class attribute ?

### DISCUSSION THREAD

**Discussion by skylerknight** [markdoc/markdoc#165] [Q&A] [answered]
Is it possible to create a custom image node with attributes other than alt,src?

It would be useful to be able to add custom classes/ids to images similar to the way it is done with heading/paragraph nodes. Like so:

```
{% .overline %} Overline paragraph

{% .full-width %} ![My Image](/assets/my-image.png)
```

**Comment by mfix-stripe** [comment, marked as answer]
This is a great thought: the Markdoc team definitely wants to support this somehow. 

One idea we had was to support attributes within the image node, like this:
```md
![My Image](/assets/my-image.png width=40 .full-width)
```

Cc @rpaul-stripe for visibility

**Reply by mfix-stripe** [reply]
@skylerknight we have an issue for tracking this support: https://github.com/markdoc/markdoc/issues/156

**Reply by skylerknight** [reply]
Thanks for the quick response.

I think that would be a decent place for attributes.

Something like this might also be easy to understand:

```md
!{% .full-width %}[My Image](assets/my-image.png)
```
